plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMaterializedViewTest.java

```diff
@@ -938,6 +938,118 @@ public void testMaterializedViewCreatedFromTableFunctionWithGracePeriod()
         assertThat(result4).isNotEqualTo(result3);
     }
 
+    @Test
+    public void testIncrementalRefresh()
+    {
+        Session defaultSession = getSession();
+        Session incrRefreshDisabledSession = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", "incremental_refresh_enabled", "false")
+                .build();
+
+        String sourceTable = "source_table_" + randomNameSuffix();
+        String matViewDef = "SELECT a, b FROM " + sourceTable + " WHERE a < 3 OR a > 5";
+
+        // create source table and two identical MVs
+        assertUpdate("CREATE TABLE " + sourceTable + " (a int, b varchar)");
+        assertUpdate("INSERT INTO " + sourceTable + " VALUES (1, 'abc'), (2, 'def')", 2);
+        String matView1 = "mat_view_test_" + randomNameSuffix();
+        String matView2 = "mat_view_test_" + randomNameSuffix();
+        assertUpdate(("CREATE MATERIALIZED VIEW " + matView1 + " AS %s").formatted(matViewDef));
+        assertUpdate(("CREATE MATERIALIZED VIEW " + matView2 + " AS %s").formatted(matViewDef));
+
+        // check MV props
+        Map<String, String> matViewProps = getStorageTableMetadata(matView1).properties();
+        assertThat(matViewProps).containsOnlyKeys("write.format.default", "write.parquet.compression-codec");
+
+        // execute first refresh: afterwards both MVs will contain: (1, 'abc'), (2, 'def')
+        assertUpdate("REFRESH MATERIALIZED VIEW " + matView1, 2);
+        assertUpdate("REFRESH MATERIALIZED VIEW " + matView2, 2);
+
+        // add some new rows to source
+        assertUpdate("INSERT INTO " + sourceTable + " VALUES (3, 'ghi'), (4, 'jkl'), (5, 'mno'), (6, 'pqr')", 4);
+
+        // will do incremental refresh, and only add: (6, 'pqr')
```

Comment:
what will happen if some past row get updated and no longer matches predicate?

Reply:
As discussed on slack, we can only do incremental refresh when delete files/overwrites are absent. IcebergSplitManager can perform this check and fall back to a full refresh in case deletes are present.